src/app/GitUI/UserControls/FileStatusList.Toolbar.cs

```diff
@@ -259,7 +275,9 @@ private void UpdateToolbar()
         bool findInFilesGitGrepVisible = CanUseFindInCommitFilesGitGrep;
         btnFindInFilesGitGrep.Visible = findInFilesGitGrepVisible;
         sepOptions.Visible = findInFilesGitGrepVisible;
-        for (int itemIndex = 0; itemIndex < btnFindInFilesGitGrep.DropDown.Items.Count; ++itemIndex)
+
+        // Init the first three items, before the separator
+        for (int itemIndex = 0; itemIndex < btnFindInFilesGitGrep.DropDown.Items.Count && btnFindInFilesGitGrep.DropDown.Items[itemIndex] is ToolStripMenuItem; ++itemIndex)
         {
             ((ToolStripMenuItem)btnFindInFilesGitGrep.DropDown.Items[itemIndex]).Checked = AppSettings.FileStatusFindInFilesGitGrepTypeIndex.Value == itemIndex;
         }
```

Comment:
`(IndexOf(sepFindUsingSettings) + 1)` needs to be applied as start `itemIndex` and as offset in the condition.

Reply:
I added an array with the order in settings with a comment